src/ru/umeta/zharvester/ZHarvester.java

```diff
@@ -50,7 +50,7 @@ public int harvest(Query query) throws Exception {
             resultsDir = query.getStartURL();
         new File(resultsDir).mkdirs();
         int num = 1000;
-        if (!query.getReg().equals("") && query.getReg().equals(null))
+        if (query.getReg() == null && !query.getReg().equals(""))
```

Comment:
I don't understand logic here. I think what you wanted was to check that is **not** null and not an empty string.